sdks/js/packages/core/react/components/organization/billing/upcoming-billing-cycle.tsx

```diff
@@ -29,6 +34,45 @@ function LabeledBillingData({
   );
 }
 
+function PlanSwitchButton({ nextPlan }: { nextPlan: V1Beta1Plan }) {
+  const intervalName = getPlanIntervalName(nextPlan).toLowerCase();
+
+  const navigate = useNavigate({ from: '/billing' });
+  function onClick() {
+    navigate({
+      to: '/billing/cycle-switch/$planId',
+      params: {
+        planId: nextPlan.id || ''
+      }
+    });
+  }
+
+  return (
+    <div>
+      <Button
+        variant={'secondary'}
+        className={billingStyles.linkBtn}
+        onClick={onClick}
+      >
+        Switch to {intervalName}
+      </Button>
+    </div>
+  );
+}
+
+function getSwitchablePlan(plans: V1Beta1Plan[], currentPlan: V1Beta1Plan) {
```

Comment:
In the original requirements, are we supposed to show the "Switch to ..." button only for upgrades, or for any cycle switch?

Reply:
i think we are showing switch for all cases. (upgrade and downgrade)

Reply:
Can you confirm on pe-design once? From the design, I also feel we need to show it for both. But based on recent experience, I think we should confirm it once. There is scope for misunderstanding there.